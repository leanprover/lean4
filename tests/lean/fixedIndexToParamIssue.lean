inductive Tree (β : Type v) where
  | leaf
  | node (left : Tree β) (key : Nat) (value : β) (right : Tree β)
  deriving Repr

inductive ForallTree (p : Nat → β → Prop) : Tree β → Prop
  | leaf : ForallTree p .leaf
  | node :
     ForallTree p left →
     p key value →
     ForallTree p right →
     ForallTree p (.node left key value right)

inductive BST : Tree β → Prop
  | leaf : BST .leaf
  | node :
     ForallTree (fun k v => k < key) left →
     ForallTree (fun k v => key < k) right →
     BST left → BST right →
     BST (.node left key value right)

#print BST
