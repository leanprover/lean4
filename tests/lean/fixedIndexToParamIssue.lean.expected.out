inductive BST.{u_1} : {β : Type u_1} → Tree β → Prop
number of parameters: 1
constructors:
BST.leaf : ∀ {β : Type u_1}, BST Tree.leaf
BST.node : ∀ {β : Type u_1} {key : Nat} {left right : Tree β} {value : β},
  ForallTree (fun k v => k < key) left →
    ForallTree (fun k v => key < k) right → BST left → BST right → BST (Tree.node left key value right)
